make-release.sh

```diff
@@ -27,132 +67,185 @@ Usage: $0 --version [VERSION TO RELEASE]
-fi
+gitClone() {
+  if [[ ${USE_TMP_DIR} -eq 1 ]]; then
+    # Getting a local copy with all ta
+    cd /tmp/
+    tmpdir=tmp-${0##*/}-$VERSION
+    rm -rf "$tmpdir"
+    git clone $REPO "$tmpdir"
+    cd "/tmp/$tmpdir"
+    git pull
+  fi
+}
+
+gitBranch() {
+  # Handle the version branch: create it or update it.
+  # Check if the branch exists, locally or on the remote.
+  EXISTING_BRANCH=0
+  git fetch
+  EXISTING_BRANCH=$(git ls-remote --heads origin "${TARGET_BRANCH}")
+  case ${EXISTING_BRANCH} in
+    "") echo "[INFO] Creating new branch: ${TARGET_BRANCH} from branch: ${MAIN_BRANCH}."
+      git checkout "${MAIN_BRANCH}"
+      git checkout -b "${TARGET_BRANCH}"
+      ;;
+    *) echo "[INFO] Updating branch: ${TARGET_BRANCH} from branch: ${MAIN_BRANCH}."
+      git checkout "${TARGET_BRANCH}"
+      git pull --rebase origin "${MAIN_BRANCH}"
+      ;;
+  esac
+}
+
+gitCommit() {
+  if  [[ ${docommit} -eq 1 ]]; then 
+    git add "$yamlfile" "modules/installation-guide/examples/checluster-properties.adoc"  "modules/installation-guide/examples/system-variables.adoc"
+    git commit -s -m "release: Bump version to ${VERSION}"
+  fi
+}
+
+gitPush() {
+  if  [[ ${dopush} -eq 1 ]]; then 
+    git pull origin "${TARGET_BRANCH}"
+    git push origin "${TARGET_BRANCH}"
+    case ${TARGET_BRANCH} in
+      "release-${VERSION}")
+        lastCommitComment="$(git log -1 --pretty=%B)"
+        hub pull-request --force --message "${lastCommitComment}" --base "${TARGET_BRANCH}" --head  "${MAIN_BRANCH}"
+        ;;
+    esac
+  fi
+}
+
+gitPullRequest() {
+  if  [[ ${docommit} -eq 1 ]]; then 
+    git pull origin "${TARGET_BRANCH}"
+    git push origin "${TARGET_BRANCH}"
+    lastCommitComment="$(git log -1 --pretty=%B)"
+    hub pull-request --force --message "${lastCommitComment}" --base "${TARGET_BRANCH}" --head  "${MAIN_BRANCH}"
+  fi
+}
 
-# where in other repos we have a VERSION file, here we have an antora-playbook.yml file which contains some keys:
-#    prod-prev-ver-major: 6 [never changes]
-#    prod-ver-major: 7 [never changes]
-#    prod-prev-ver: 7.24 [always prod-ver - 1]
-#    prod-ver: 7.25
-#    prod-ver-patch: 7.25.2
-playbookfile=antora-playbook.yml
-updateYaml() {
-  NEWVERSION=${1}
-  echo "[INFO] update $playbookfile with prod-ver = $NEWVERSION"
-
-  [[ $NEWVERSION =~ ^([0-9]+)\.([0-9]+)\.([0-9]+) ]] && BASE1="${BASH_REMATCH[1]}"; BASE2="${BASH_REMATCH[2]}"; # for VERSION=7.25.2, get BASE1=7; BASE2=25
-  # prod-ver should never go down, only up
-  OLDVERSION=$(cat $playbookfile | yq -r '.asciidoc.attributes."prod-ver-patch"') # existing prod-ver-patch version 7.yy.z
-  VERSIONS="${OLDVERSION} ${NEWVERSION}"
-  VERSIONS_SORTED="$(echo $VERSIONS | tr " " "\n" | sort -V | tr "\n" " ")"
+gitTag() {
+  if [[ $TAG_RELEASE -eq 1 ]]; then
+    echo "[INFO] Creating release tag"
+    git checkout "${TARGET_BRANCH}"
+    git pull origin "${TARGET_BRANCH}" || true
+    git tag "${VERSION}"
```

Comment:
any support for if tag already exists?

Reply:
Then the release already exists, and we should already have failed when checking if version is incremented.